lib/src/rules/sort_child_properties_last.dart

```diff
@@ -108,31 +108,27 @@ class _Visitor extends SimpleAstVisitor {
     }
 
     var arguments = node.argumentList.arguments;
-    if (arguments.length < 2 ||
-        isChildArg(arguments.last) ||
-        arguments.where(isChildArg).length != 1) {
+    if (arguments.length < 2 || isWidgetProperty(arguments.last.staticType)) {
       return;
     }
 
-    var onlyClosuresAfterChild = arguments.reversed
-        .takeWhile((argument) => !isChildArg(argument))
-        .toList()
-        .reversed
-        .where((element) =>
-            element is NamedExpression &&
-            element.expression is! FunctionExpression)
-        .isEmpty;
-    if (!onlyClosuresAfterChild) {
-      rule.reportLint(arguments.firstWhere(isChildArg));
+    var reversedArguments = arguments.reversed.toList();
+    var lastWidget = reversedArguments
+        .where((e) => isWidgetProperty(e.staticType))
+        .firstOrNull;
```

Comment:
Seem like this would be more efficient if it used `lastIndexWhere`. We wouldn't need to build `reversedArguments` and we could just iterate from `index` to `length` below when looking for a non-closure rather than build iterators. But maybe iterators are more performant than I'm guessing they are. No need to change this if you're reasonably confident that it wouldn't provide a performance boost.

Reply:
Thanks for the suggestion. Done.